lib/chef/config.rb

```diff
@@ -569,12 +569,8 @@ def self.env
       ENV
     end
 
-    def self.windows_home_path
```

Comment:
There's a complex framework that we used for deprecating some methods for Chef 12 that you can see in ```lib/chef/mixin/deprecation.rb``` but the simple thing is ```Chef::Log.deprecation("blah blah blah is blah because deprecated blah")```.